network/src/gossip.rs

```diff
@@ -385,6 +384,10 @@ impl NewLeafActions {
 #[derive(Clone)]
 pub struct RegisteredMessageValidator {
 	inner: Arc<MessageValidator<dyn ChainContext>>,
+	// Note: this is always `Some` in real code and `None` in tests.
+	service: Option<Arc<PolkadotNetworkService>>,
+	// Note: this is always `Some` in real code and `None` in tests.
```

Comment:
This really isn't great, but I couldn't come up with a better solution.

Reply:
Hide it under a `#[cfg(not(test))]` flag perhaps?